### PR TITLE
Default capture options

### PR DESCRIPTION
**Pull request by hiroshihorie**
LK-319
Simply pass default capture settings.
Same as https://github.com/livekit/client-sdk-ios/pull/32

### CODE REVIEW

File: lib/src/participant/local_participant.dart [hunk 1]
```diff
@@ -296,13 +296,16 @@ class LocalParticipant extends Participant<LocalTrackPublication> {
...
         return await publishVideoTrack(track);
       } else if (source == TrackSource.microphone) {
-        final track = await LocalAudioTrack.create();
+        final track = await LocalAudioTrack.create(
+            room.roomOptions?.defaultAudioCaptureOptions);
         return await publishAudioTrack(track);
       } else if (source == TrackSource.screenShareVideo) {
-        final track = await LocalVideoTrack.createScreenShareTrack();
+        final track = await LocalVideoTrack.createScreenShareTrack(
+            room.roomOptions?.defaultVideoCaptureOptions);
```
Comment (boks1971) [inline lib/src/participant/local_participant.dart:308]: Are these the same in Flutter case? I am guessing we are doing Simulcast for camera and non-Simulcast for screen share. Can Flutter do that?

File: lib/src/track/local/video.dart [hunk 2]
```diff
@@ -49,9 +53,13 @@ class LocalVideoTrack extends LocalTrack with VideoTrack {
...
   static Future<LocalVideoTrack> createScreenShareTrack([
-    ScreenShareTrackOptions? options,
+    VideoCaptureOptions? options,
   ]) async {
-    options ??= const ScreenShareTrackOptions();
+    if (options is VideoCaptureOptions &&
+        options is! ScreenShareCaptureOptions) {
+      options = ScreenShareCaptureOptions.from(captureOptions: options);
+    }
+    options ??= const ScreenShareCaptureOptions();
```
Comment (boks1971) [inline lib/src/track/local/video.dart:62]: Oh looks like this is overriding default options if it is not screen share options. Seems a bit confusing. Is it possible to just set the defaults to the right values in the caller?

File: lib/src/participant/local_participant.dart (hunk 1 above)
Comment (hiroshihorie) [reply lib/src/participant/local_participant.dart:308]: Makes sense to separate defaults for cam and screen share 😁

**Comment by hiroshihorie** [comment]
Sounds like splitting default options for Camera and Screen share is better 🙂

**Comment by hiroshihorie** [comment]
Separate camera & screen share capture options 🙂

File: lib/src/track/local/video.dart (hunk 2 above)
Comment (hiroshihorie) [reply lib/src/track/local/video.dart:55]: I'm trying to keep it consistent with this pattern 👉 https://github.com/livekit/client-sdk-flutter/blob/main/lib/src/participant/local_participant.dart#L88-L100

For now .. but I think we should move the create track methods to
`LocalParticipant.createCameraTrack` so there is access to the default room options.

**Review by davidliu — APPROVED** [review]
LGTM

**Pull request merged**